tests/test_sig_stfl_key_save.c

```diff
@@ -0,0 +1,304 @@
+// SPDX-License-Identifier: MIT
+
+#if defined(_WIN32)
+#pragma warning(disable : 4244 4293)
+#endif
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <oqs/oqs.h>
+
+#if OQS_USE_PTHREADS_IN_TESTS
+#include <pthread.h>
+#endif
+
+#ifdef OQS_ENABLE_TEST_CONSTANT_TIME
+#include <valgrind/memcheck.h>
+#define OQS_TEST_CT_CLASSIFY(addr, len)  VALGRIND_MAKE_MEM_UNDEFINED(addr, len)
+#define OQS_TEST_CT_DECLASSIFY(addr, len)  VALGRIND_MAKE_MEM_DEFINED(addr, len)
+#else
+#define OQS_TEST_CT_CLASSIFY(addr, len)
+#define OQS_TEST_CT_DECLASSIFY(addr, len)
+#endif
+
+#include "system_info.c"
+
+typedef struct magic_s {
+	uint8_t val[31];
+} magic_t;
+
+#define MAX_FILE_NAME_LEN 60
+char filename[MAX_FILE_NAME_LEN] = "XMSS-SHA2_10_256";
+
+void prepend(char *s, const char *t) {
+	size_t len = strlen(t);
+	memmove(s + len, s, strlen(s) + 1);
+	memcpy(s, t, len);
+}
+
+OQS_STATUS lock_sk_key(OQS_SECRET_KEY *sk) {
+	printf("lk=%02x\n", sk->secret_key[0]);
+	return 0;
+}
+
+OQS_STATUS release_sk_key(OQS_SECRET_KEY *sk) {
+	printf("rk=%02x\n", sk->secret_key[0]);
+	return 0;
+}
+
+OQS_STATUS do_nothing_save(const OQS_SECRET_KEY *sk) {
+	printf("save_k=%02x\n", sk->secret_key[0]);
+	return 0;
+}
+
+OQS_STATUS sk_file_write(const OQS_SECRET_KEY *sk) {
+
+	prepend(filename, "./tmp/oqs_");
+	strcat(filename, "__sk");
+
+	printf("\nWriting to file %s\n", filename);
+	FILE *printer = fopen(filename, "w+");
+	if (printer == NULL) {
+		perror("ERROR! There is no such file. Terminating...");
+		return -1;
+	}
+
+	// Write the entire secret key byte array to the specified file.
+	for (unsigned long i = 0; i < sk->length_secret_key; i++) {
```

Comment:
```suggestion
	for (size_t i = 0; i < sk->length_secret_key; i++) {
```

Reply:
The entire file has been deleted because it is now redundant.